### PR TITLE
fix: retrieve publickey resources by labels

### PR DESCRIPTION
**Pull request by fra98**
# Description

This PR fixes a bug in Liqoctl where PublicKey resources were retrieved by name instead of labels.

### CODE REVIEW

File: pkg/liqoctl/network/cluster.go [hunk 1]
```diff
@@ -322,8 +322,18 @@ func (c *Cluster) EnsureGatewayClient(ctx context.Context, opts *forge.GwClientO
...
-		remoteClusterID, key)
+
+	// Check if the PublicKey already exists.
+	var name *string
+	pk, err := getters.GetPublicKeyByClusterID(ctx, c.local.CRClient, remoteClusterID)
+	if client.IgnoreNotFound(err) != nil {
+		s.Fail(fmt.Sprintf("An error occurred while retrieving public key: %v", output.PrettyErr(err)))
+		return err
+	} else if err == nil {
+		name = &pk.Name // reutilize the existing PublicKey name
```
Comment (claudiolor) [inline pkg/liqoctl/network/cluster.go:333]: To make things clearer I would say something like:
```suggestion
		name = &pk.Name // if the public key already exists, keep its name
```

**Review by claudiolor — APPROVED** [review]

File: pkg/liqo-controller-manager/networking/forge/publickey.go [hunk 2]
```diff
@@ -29,20 +30,20 @@ import (
...
-// DefaultPublicKeyName returns the default name of a PublicKey.
-func DefaultPublicKeyName(remoteClusterID liqov1beta1.ClusterID) string {
+// defaultPublicKeyName returns the default name of a PublicKey.
+func defaultPublicKeyName(remoteClusterID liqov1beta1.ClusterID) string {
 	return string(remoteClusterID)
 }
 
 // PublicKey forges a PublicKey.
-func PublicKey(name, namespace string, remoteClusterID liqov1beta1.ClusterID, key []byte) (*networkingv1beta1.PublicKey, error) {
+func PublicKey(namespace string, name *string, remoteClusterID liqov1beta1.ClusterID, key []byte) (*networkingv1beta1.PublicKey, error) {
```
Comment (aleoli) [inline pkg/liqo-controller-manager/networking/forge/publickey.go:39]: I'm just curious: why did you invert the order?

**Review by aleoli — APPROVED** [review]

**Pull request merged**